### PR TITLE
Insert and match inherited roles tests

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
Extend a scenario to insert and match against a relation that only has inherited roles.

## What are the changes implemented in this PR?
* Note: both the insert, and the match may fail

### CODE REVIEW

File: graql/language/rule-validation.feature [hunk 1]
```diff
@@ -19,7 +19,7 @@ Feature: Graql Rule Validation
 
   Background: Initialise a session and transaction for each scenario
     Given connection has been opened
-    Given connection open session for databases: test_rule_validation
+    Given connection open schema session for database: test_rule_validation
```
Comment (flyingsilverfin) [inline graql/language/rule-validation.feature:22]: ignore this, just saw it and wanted to fix it, but probably incomplete

File: graql/language/match.feature [hunk 2]
```diff
@@ -1152,14 +1154,19 @@ Feature: Graql Match Query
     Then answer size is: 2
     When get answers of graql query
       """
-      match (spouse: $x, spouse: $y) isa marriage;
+      match (spouse: $x, spouse: $y) isa civil-marriage;
```
Comment (flyingsilverfin) [inline graql/language/match.feature:1157]: this is the new match

File: graql/language/match.feature [hunk 3]
```diff
@@ -1126,6 +1127,7 @@ Feature: Graql Match Query
       $a isa person, has ref 1;
       $b isa person, has ref 2;
       (wife: $a, husband: $b) isa hetero-marriage;
+      (spouse: $a, spouse: $b) isa civil-marriage;
```
Comment (flyingsilverfin) [inline graql/language/match.feature:1130]: this is a new insert

File: graql/language/match.feature [hunk 4]
```diff
@@ -1114,6 +1114,7 @@ Feature: Graql Match Query
       person plays hetero-marriage:husband, plays hetero-marriage:wife;
       marriage sub relation, relates spouse;
       hetero-marriage sub marriage, relates husband as spouse, relates wife as spouse;
+      civil-marriage sub marriage;
```
Comment (flyingsilverfin) [inline graql/language/match.feature:1117]: this new type of marriage only has inherited roles

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**